### ISSUE TITLE
[Question] Why use scrollHeight in particular?

### ISSUE THREAD

**Issue by Titozzz** [roginfarrer/react-collapsed#31]
This lib gets children size with scrollHeight but it can cause issues if that child contains something with a dropdown that would not be displayed but would still count towards scrollHeight.
We found workarounds in our use cases but we were wondering why this choice?
There are some alternatives: https://medium.com/@jbbpatel94/difference-between-offsetheight-clientheight-and-scrollheight-cfea5c196937

🚀

**Comment by roginfarrer** [comment]
Quite frankly I struggle to grasp the difference between all of them. If I recall correctly I tried each of them in the library, and landed with scrollHeight since it seemed to work without any issues. I'm open to a PR to change it if we can document a good reason for the change.

**Comment by roginfarrer** [comment]
Ah, I think I remember. When debugging #14, I discovered that when padding is applies to the collapse element, `scrollHeight` is the only measurement that is accurate. In this example, I changed the measurement to `clientHeight`, and it ends the transition prematurely. 

See screenshot:

<img width="1387" alt="Screen Shot 2019-09-26 at 10 13 49 PM" src="https://user-images.githubusercontent.com/9063669/65737074-4e591080-e0ab-11e9-8942-94d4a318be8f.png">
